amazon-kinesis-client/src/test/java/software/amazon/kinesis/lifecycle/ShardConsumerSubscriberTest.java

```diff
@@ -311,6 +309,80 @@ public void restartAfterRequestTimerExpiresTest() throws Exception {
 
     }
 
+    @Test
+    public void restartAfterRequestTimerExpiresAfterInitialSubscriptionFailsTest() throws Exception {
```

Comment:
The test case is hard to follow and has lot of waits. We need to avoid sleeps. adding comments would be helpful. We can sync up offline to check if the test can be shortened.

Reply:
I have refactored the unit test and added comments to made it easier to read. It takes for it to 60 to 80 ms to run.